### ISSUE TITLE
How to install pgbackrest with init cluster ?

### ISSUE THREAD

**Issue by karnain1803** [vitabaks/postgresql_cluster#632]
Hi, what parameters should I change in order to simply install along with the pgbackrest cluster and it didn't fall into error? When I set pgbackrest_install: true, it waits for the command to restore the database from the backup, but I just need to install this along with the cluster.

**Comment by karnain1803** [comment]
and for what in inventory file has pgbackrest:vars ?

**Comment by vitabaks** [comment]
>  what parameters should I change in order to simply install along with the pgbackrest cluster

```
patroni_cluster_bootstrap_method: "initdb"
patroni_create_replica_methods:
  - basebackup
pgbackrest_install: true
```

>  for what in inventory file has pgbackrest:vars ?

To configure the dedicated repository host (optional, if pgbackrest_install: true and "repo_host" is set).

**Comment by karnain1803** [comment]
why patroni_create_peplica_methods choosing on the basebackup, but no pgbackrest ? if pgbackrest too replica method for create ?

**Comment by vitabaks** [comment]
If I understood correctly, the first question was how to simply install pgbackrest. If this is the case, `pgbackrest_install: true` and the default values for the rest of the variables are sufficient. 

Everything else depends on the task you are currently solving, most variables can be redefined (if you know what you are doing).

**Comment by karnain1803** [comment]
> If I understood correctly, the first question was how to simply install pgbackrest package. If this is the case, `pgbackrest_install: true` and the default values for the rest of the variables are sufficient.
> 
> Everything else depends on the task you are currently solving, most variables can be redefined (if you know what you are doing).

Don't you think it's cool to add a playbook for backup using pgbackrest? In storage, on a remote server and locally?

**Comment by vitabaks** [comment]
> Don't you think it's cool to add a playbook for backup using pgbackrest? In storage, on a remote server and locally?

This has already been implemented. See the pgbackrest [variables](https://github.com/vitabaks/postgresql_cluster/blob/8e7151c2227183bb80f6f81147f8ecde3e656c6f/vars/main.yml#L505).

**Comment by karnain1803** [comment]
I want to restore the cluster by a certain time, when I log in to the master node, I can't do this, since I suppose I need to turn off the entire cluster? How could I do a proper database data recovery, and that the recovery would also be successful on replicas?

**Comment by vitabaks** [comment]
Please see README https://github.com/vitabaks/postgresql_cluster?tab=readme-ov-file#restore-and-cloning

“Point-In-Time-Recovery” section.

**Comment by karnain1803** [comment]
Yes, I looked at it. This is going to sound silly, but could you please clarify one thing for me. I have a backup on the host of the master node, when I make the setting that you specified for PITR, I get this. https://pastebin.com/JpY5GQHi That's why I'm asking, what's the right way to do it?

**Comment by karnain1803** [comment]
I will correct the question, how to make a backup on all nodes at once?

**Comment by vitabaks** [comment]
Setting up backups locally is a bad idea, especially if it's a production cluster. 

 Use a dedicated backup server or s3.

**Comment by karnain1803** [comment]
Yes, you`re right. This is only test. Just i want to know "how to make a backup on all nodes at once" ? This is not provided in your playbook, except for the crone

**Comment by vitabaks** [comment]
This is an open source project, I use it personally and distribute it as is… \
I have plans to add more detailed documentation, I need to take more time for this.

 You can also review the sponsorship program, which includes individual support.

**Comment by karnain1803** [comment]
Thank you sir
I have one last question. Is it meant to manually configure the user for dedicated host https://github.com/vitabaks/postgresql_cluster/blob/c12182611d6e610ce6b6d938ea813c4d2fbc4e3e/inventory#L61 Or is it enough for me to set pgbackrest_repo_host: "IP-Remote" without a user?  Or I exhibit in vars/main.xml repo_host and repo_user and will such a user be created automatically?

**Comment by vitabaks** [comment]
Yes, you can redefine the values of these variables to suit your environment.

**Comment by mrkhachaturov** [comment]
@vitabaks 
After cluster init  do we need to change `patroni_create_replica_methods` configuration?
I mean do we need to enable `pgbackrest` and if so , how to run command, to apply this changes  in cli after init?
```
patroni_create_replica_methods:
  - pgbackrest
  #  - wal_g
  #  - pg_probackup
  # - basebackup
```

**Comment by vitabaks** [comment]
Use config_pgcluster.yml to apply changes.

Doc: https://autobase.tech/docs/management/configuration
